### ISSUE TITLE
[FEAT] Improve Firewall

### ISSUE THREAD

**Issue by monsieuremre** [secureblue/secureblue#294]
Configuring firewalld daemon I see. Might be a good idea when you are the user. But when creating an ISO, there are more direct and effective ways. Why not just literally modify the kernel nftables config? Like push the following rules for example:
```
table inet filter {
   # Block everything incoming
    chain input {
        type filter hook input priority 0; policy drop;
    }

   # Accept outgoing
    chain output {
        type filter hook output priority 0; policy accept;
        ct state established,related accept
    }
   
    # Block routing
    chain forward {
        type filter hook forward priority 0; policy drop;
    }
}

```
Just running nft 'rule' will push the rule to the system config or just edit the config directly. You wanna go further in? Maybe block some ports even for outgoing that are for unencrypted and obsolete protocols. I don't know, get creative. But this is the more direct and better way, better than depending on a system daemon.

**Comment by RoyalOughtness** [comment]
This project doesn't create an iso. Also, why would we modify a lower level config when a cleaner abstraction is available? 

This issue makes no sense.

**Comment by monsieuremre** [comment]
> This project doesn't create an iso. 

My bad with the terminology then.

> Also, why would we modify a lower level config when a cleaner abstraction is available?

Because it is a highly sophisticated ever running daemon with elevated privileges that can directly interract and modify the system, that is made that way so that the user can jump between zones in different networks with. If you are planning to stick to one zone all the time, then using firewalld just adds more unnecessary attack surface. Thats pretty much it. If user modifiability is a concern, then you are right, this is the better way. But if you plan on hardcoding most configs, doing it at low level and stripping all user and system daemons of the privileges to modify network firewall settings with a systemd dropin file is a much, much secure option, obviously. TL;DR=less surface and really 'not easy' to modify as the user or any malicious script.

> This issue makes no sense.

Not the best choice of words. Might just say it is out of scope or not planned. Depending on what this project actually aims, it might make a lot of sense.

**Comment by TommyTran732** [comment]
So I got a question - when something like what you are suggested is implemented, can't scripts running as root still modify the firewall? Because it really sounds like security by obscurity.

**Comment by RoyalOughtness** [comment]
@monsieuremre 

>  If you are planning to stick to one zone all the time, then using firewalld just adds more unnecessary attack surface.

Users that plan to use one zone and know how to use nftables can do that then. Otherwise, disabling firewalld sounds like a recipe for less knowledgable users to unintentionally open ports.

> Because it is a highly sophisticated ever running daemon 

Sure, it's one of many. 


> stripping all user and system daemons of the privileges

If you're interested in helping harden and isolate system services, check out https://fedoraproject.org/wiki/Changes/SystemdSecurityHardening

Once this lands in Fedora 41 it will have a much broader impact than focusing for some reason on one specific service.

>  much, much secure option, obviously.

I agree with @TommyTran732. This is not as "obvious" as you think it is especially when upstream is actively working on service hardening and isolation.

**Comment by monsieuremre** [comment]
@TommyTran732 of course anyone with root is playing in godmode. But that was not the point.

> Because it really sounds like security by obscurity.

Not at all. It is security by attack surface reduction.


Also plus if the second step is done, the reduction would be significant. Because if you disable network firewall privileges for all systemd units with a drop in, then even if a daemon elevates privileges, it won't be useful.

> If you're interested in helping harden and isolate system services, check out https://fedoraproject.org/wiki/Changes/SystemdSecurityHardening

If mainstream actually gets some of these changes, that would be a huge improvement. Hope it goes through. Will look into it.

**Comment by RoyalOughtness** [comment]
@monsieuremre It almost certainly will. It just missed the F40 cutoff: https://github.com/coreos/fedora-coreos-tracker/issues/1662#issuecomment-2020994529

**Comment by RoyalOughtness** [comment]
https://docs.google.com/spreadsheets/d/1BLKqBSsF0B9gYz6b4TIPnJ93SpiUVDiD8z_80pTyXJc/edit?gid=0#gid=0

**Comment by monsieuremre** [comment]
> @monsieuremre It almost certainly will. It just missed the F40 cutoff: https://github.com/coreos/fedora-coreos-tracker/issues/1662#issuecomment-2020994529

This is very interesting. As of now, no linux distro is actually 'good' in terms of default security at all. This would make fedora stand out a bit.

**Comment by RoyalOughtness** [comment]
@monsieuremre yet another reason to use fedora as a base for a hardened system :smile:
